### PR TITLE
[FIX] purchase_stock: UoM rounding

### PR DESCRIPTION
**Pull request by nim-odoo**
opw-2046965

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by nim-odoo** [comment]
robodoo r+

File: addons/purchase_stock/tests/test_create_picking.py [hunk 1]
```diff
@@ -313,3 +313,65 @@ def test_04_mto_multiple_po(self):
...
         self.assertEqual(sum(customer_picking.move_lines.mapped('reserved_availability')), 100.0, 'The total quantity for the customer move should be available and reserved.')
+
+    def test_04_rounding(self):
+        """ We set the Unit(s) rounding to 1.0 and ensure buying 1.2 units in a PO is rounded to 1.0
+            at reception.
+        """
+        uom_unit = self.env.ref('uom.product_uom_unit')
+        uom_unit.rounding = 1.0
+
+        # buy a dozen
```
Comment (madprog) [inline addons/purchase_stock/tests/test_create_picking.py]: Shouldn't that be `# buy 1.2 dozen`?

File: addons/purchase_stock/tests/test_create_picking.py [hunk 2]
```diff
@@ -313,3 +313,65 @@ def test_04_mto_multiple_po(self):
...
+        po.button_confirm()
+
+        # the move should be 1.0 units
+        move1 = po.picking_ids.move_lines[0]
+        self.assertEqual(move1.product_uom_qty, 1.0)
+        self.assertEqual(move1.product_uom.id, uom_unit.id)
+        self.assertEqual(move1.product_qty, 1.0)
+
+        # edit the so line, buy 2.4 units, the move should now be 2.0 units
+        po.order_line.product_qty = 2.0
```
Comment (madprog) [inline addons/purchase_stock/tests/test_create_picking.py]: Shouldn't the test assign `2.4` instead?

Comment (nim-odoo) [reply addons/purchase_stock/tests/test_create_picking.py]: Too bad I saw this comment too late.

**Pull request closed without merging**